### PR TITLE
[FIX] website_gengo: Now working properly as in v8

### PR DESCRIPTION
**Pull request by hmo-odoo**
```
- Fixed linking issue after 'Translate' button click.
- Controller added to catch gengo post as in v8
```

### CODE REVIEW

File: addons/website_gengo/static/src/css/website_gengo.css [hunk 1]
```diff
@@ -1,3 +1,3 @@
-.o_translatable_inprogress {
```
Comment (hmo-odoo) [inline addons/website_gengo/static/src/css/website_gengo.css]: do not remove css class.

File: addons/website_gengo/static/src/css/website_gengo.css [hunk 2]
```diff
@@ -1,3 +1,3 @@
...
-  background: #b7e4ff;
-}
\ No newline at end of file
+html[lang] > body.editor_enable [data-oe-translation-state="inprogress"] {
```
Comment (hmo-odoo) [inline addons/website_gengo/static/src/css/website_gengo.css]: do not try to override css property using important.

you should add o_translatable_inporgress class when changed oe_translation-state data value.

File: addons/website_gengo/controllers/main.py [hunk 3]
```diff
@@ -37,3 +37,30 @@ def set_gengo_config(self,config):
     def post_gengo_jobs(self):
         request.registry['base.gengo.translations']._sync_request(request.cr, request.uid, limit=GENGO_DEFAULT_LIMIT, context=request.context)
         return True
+
+    @http.route('/website/set_translations', type='json', auth='user', website=True)
```
Comment (msh-odoo) [inline addons/website_gengo/controllers/main.py]: Either this controller should be moved to website or if it is used only for website_gengo then route should start with website_gengo

File: addons/website_gengo/controllers/main.py [hunk 4]
```diff
@@ -37,3 +37,30 @@ def set_gengo_config(self,config):
...
+    def set_translations(self, data, lang):
+        IrTranslation = request.env['ir.translation']
+        for term in data:
+            initial_content = term['initial_content'].strip()
+            tid = term['translation_id']
+            if not tid:
+                translations = IrTranslation.search_read([('lang', '=', lang), ('src', '=', initial_content)], fields=['id'])
+                if translations:
+                    tid = map(lambda t_id: t_id['id'], translations)
```
Comment (msh-odoo) [inline addons/website_gengo/controllers/main.py]: You may get more than one record for same src, what if you get more then one record for same source, should all those entries updated(if yes then shouldn't it be translation_ids) ?

File: addons/website_gengo/static/src/css/website_gengo.css (hunk 1 above)
Comment (vbh-odoo) [inline addons/website_gengo/static/src/css/website_gengo.css]: this class is now useless, now we uses ```[data-oe-translation-state]``` instead.

File: addons/website_gengo/static/src/css/website_gengo.css (hunk 2 above)
Comment (vbh-odoo) [inline addons/website_gengo/static/src/css/website_gengo.css]: Its parent already uses ```!important```, to override the effect of its parent, we also required to use ```!important``` here.
```
html[lang] > body.editor_enable [data-oe-translation-state] {
  background: rgba(255, 255, 90, 0.5) !important;
}
```
https://github.com/odoo/odoo/blob/master/addons/web_editor/static/src/css/editor.css#L835

**Comment by nle-odoo** [comment]
merged in 9.0 with https://github.com/odoo/odoo/commit/8acfb0d72dd1d3f570df700e3483e639a5e4d666

**Pull request closed without merging**